doc/manual/src/development/debugging.md

```diff
@@ -0,0 +1,61 @@
+# Debugging Nix
+
+This section provides instructions on how to build and debug Nix with debug
+symbols enabled. It assumes you are using Nix with the [`flakes`] and
+[`nix-command`] experimental features enabled.
+
+[`flakes`]: @docroot@/development/experimental-features.md#xp-feature-flakes
+[`nix-command`]: @docroot@/development/experimental-features.md#xp-nix-command
+
+## Building Nix with Debug Symbols
+
+First, ensure you have set up the development environment as described in the
+[building documentation](./building.md).
+
+In the development shell, set the `mesonBuildType` environment variable to
+`debug` before configuring the build:
+
+```console
+[nix-shell]$ export mesonBuildType=debug
+```
+
+Then, configure and build Nix:
+
+```console
+[nix-shell]$ mesonConfigurePhase
+[nix-shell]$ ninjaBuildPhase
+```
+
+This will build Nix with debug symbols, which are essential for effective
+debugging.
+
+## Debugging the Nix Binary
+
+### Installing a Debugger
+
+Install your preferred debugger within the development shell. For example, to
+install `lldb`:
```

Comment:
Maybe we want to suggest gdb on Linux still by default? At least for me it used to be more stable. But I am happily convinced if this has changed. And on macOS I believe we should not suggest to install lldb with nix-shell because it's not code-signed, meaning it won't be able to use the debugger api without disabling apples security protection.

Reply:
For Linux I can see why we would want to recommend `gdb`, it's just the case that I am on Mac an that's the only thing I am familiar with.

I haven't experienced needing to disable security protection on macOS to work with lldb in nix-shell.